From 33d55b4bede31c4304be7f91c3a6936357a19273 Mon Sep 17 00:00:00 2001
From: bsapthagiri <sapthagiri.baratam@incubesol.com>
Date: Wed, 9 Jul 2014 16:19:28 +0530
Subject: [PATCH 267/351] ASoC: arizona: Reading coeff controls from DSP
 memory.

Added code to get coeff controls directly from DSP memory
instead of from cache when DSP is in running state.

Change-Id: I2d315ffc62581c45f983bf9ead9097e92edff1dd
Signed-off-by: bsapthagiri <sapthagiri.baratam@incubesol.com>
---
 sound/soc/codecs/wm_adsp.c |    5 +++++
 1 file changed, 5 insertions(+)

diff --git a/sound/soc/codecs/wm_adsp.c b/sound/soc/codecs/wm_adsp.c
index ee6dd5e..8809f26 100644
--- a/sound/soc/codecs/wm_adsp.c
+++ b/sound/soc/codecs/wm_adsp.c
@@ -665,9 +665,14 @@ static int wm_coeff_get(struct snd_kcontrol *kcontrol,
 			struct snd_ctl_elem_value *ucontrol)
 {
 	struct wm_coeff_ctl *ctl = (struct wm_coeff_ctl *)kcontrol->private_value;
+	struct wm_adsp *adsp = ctl->adsp;
 	char *p = ucontrol->value.bytes.data;
 
 	mutex_lock(&ctl->lock);
+
+	if (adsp->running)
+		wm_coeff_read_control(kcontrol, ctl->cache, ctl->len);
+
 	memcpy(p, ctl->cache, ctl->len);
 	mutex_unlock(&ctl->lock);
 	return 0;
-- 
1.7.9.5

